sql/plan/alter_foreign_key.go

```diff
@@ -33,19 +33,26 @@ var (
 )
 
 type CreateForeignKey struct {
-	BinaryNode // Left: child, Right: parent
-	FkDef      *sql.ForeignKeyConstraint
+	ddlNode
```

Comment:
Why this change? This moves the table resolution logic out of the analyzer and puts it here.

Reply:
You should add a note about why it's structured this way